docs/_renderer.py

```diff
@@ -248,9 +283,15 @@ def check_if_missing_expected_example(el, converted):
         # Only check Shiny objects for examples
         return
 
-    if hasattr(el, "decorators") and "no_example" in [
-        d.value.canonical_name for d in el.decorators
-    ]:
+    def is_no_ex_decorator(x):
+        if x == "no_example()":
+            return True
+
+        return x == f'no_example("{os.environ.get("SHINY_MODE", "core")}")'
```

Comment:
Do we need to be worried about people potentially using single-quote?

Reply:
I don't think so. Our linters won't let them and I think the string we see in quartodoc is standardized anyway.

Reply:
This can probably be improved, but I'm okay with it for now. It's not blocking and if it becomes problematic, issues would surface in CI on PRs first.